### PR TITLE
fix: allow authorization header to have different type

### PR DESCRIPTION
**Pull request by KamaniBhavin**
Enable the authorization header to support various types such as Bearer, X-Api-Key, and others. This flexibility accommodates a wide range of API providers that utilize the Authorization header with custom types.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ app.all("*", async (context) => {
   const xGatewayServiceToken = clone.headers.get("x-gateway-service-token");
   const xGatewayServiceAuthKey = clone.headers.get("x-gateway-service-auth-key");
   const xGatewayServiceAuthType = clone.headers.get("x-gateway-service-auth-type");
+  const xGatewayAuthorizationType = clone.headers.get("x-gateway-authorization-type");
```
Comment (twodayslate) [inline src/index.ts:21]: Should probably be consistent and use X-gateway-service

**Review by twodayslate — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (twodayslate) [reply src/index.ts:21]: Instead of type what about prefix? Thoughts?

File: src/index.ts [hunk 2]
```diff
@@ -18,6 +18,7 @@ app.all("*", async (context) => {
   const xGatewayServiceToken = clone.headers.get("x-gateway-service-token");
   const xGatewayServiceAuthKey = clone.headers.get("x-gateway-service-auth-key");
   const xGatewayServiceAuthType = clone.headers.get("x-gateway-service-auth-type");
+  const xGatewayAuthorizationType = clone.headers.get("x-gateway-service-authorization-type");
```
Comment (twodayslate) [inline src/index.ts:21]: Yeah I think prefix instead of type makes more sense?

**Review by twodayslate — CHANGES_REQUESTED** [review]

**Comment by twodayslate** [comment]
Got the perfect name

`x-gateway-service-auth-header-prefix`

File: src/index.ts [hunk 3]
```diff
@@ -50,7 +51,7 @@ app.all("*", async (context) => {
   const headers = new Headers(filteredHeaders);
 
   if (xGatewayServiceAuthKey && xGatewayServiceAuthType === ServiceAuthType.HEADER) {
-    const value = xGatewayServiceAuthKey.toLowerCase() === "authorization" ? `Bearer ${token}` : token;
+    const value = xGatewayAuthorizationType ? `${xGatewayAuthorizationType} ${token}` : token;
     headers.append(xGatewayServiceAuthKey, value);
   } else if (xGatewayServiceAuthKey && xGatewayServiceAuthType === ServiceAuthType.QUERY) {
     url.searchParams.append(xGatewayServiceAuthKey, token);
```
Comment (twodayslate) [inline src/index.ts:57]: Should we do the same here?

Then it should be named

`x-gateway-service-auth-prefix`

File: src/index.ts (hunk 2 above)
Comment (KamaniBhavin) [reply src/index.ts:21]: ok

File: src/index.ts (hunk 3 above)
Comment (KamaniBhavin) [reply src/index.ts:57]: are you talking about `xGatewayServiceAuthKey` ?

Comment (twodayslate) [reply src/index.ts:57]: Have the option to prefix token if it's a query as well.

**Review by twodayslate — APPROVED** [review]

**Pull request merged**